crates/vm/levm/src/opcode_handlers/arithmetic.rs

```diff
@@ -148,22 +148,25 @@ impl<'a> VM<'a> {
         let [augend, addend, modulus] = *current_call_frame.stack.pop()?;
 
         if modulus.is_zero() {
-            current_call_frame.stack.push(&[U256::zero()])?;
+            current_call_frame.stack.push1(U256::zero())?;
             return Ok(OpcodeResult::Continue { pc_increment: 1 });
         }
 
         let new_augend: U512 = augend.into();
         let new_addend: U512 = addend.into();
 
-        let sum = new_augend
-            .checked_add(new_addend)
-            .ok_or(InternalError::Overflow)?;
+        #[allow(
+            clippy::arithmetic_side_effects,
+            reason = "both values come from a u256, so the product can fit in a U512"
+        )]
+        let sum = new_augend + new_addend;
+        #[allow(clippy::arithmetic_side_effects, reason = "can't overflow")]
+        let sum_mod = sum % modulus;
 
-        let sum_mod = sum
-            .checked_rem(modulus.into())
-            .ok_or(InternalError::Overflow)?
+        #[allow(clippy::expect_used, reason = "can't overflow")]
+        let sum_mod: U256 = sum_mod
             .try_into()
-            .map_err(|_err| InternalError::Overflow)?;
+            .expect("can't fail because we applied % mod where mod is a U256 value");
```

Comment:
I think it's worth checking if using `unwrap_unchecked` gives a measurable improvement in this case. If it does, I'd replace the expect with that.